app/controllers/constants_controller.rb

```diff
@@ -0,0 +1,76 @@
+class ConstantsController < ApplicationController
+  before_action :login_required
+  before_action :verify_admin
+  before_action :set_constant, only: [:show, :edit, :update, :destroy]
+
+  # GET /constants
+  # GET /constants.json
+  def index
+    @constants = Constant.all
+  end
+
+  # GET /constants/1
+  # GET /constants/1.json
+  def show
+  end
+
+  # GET /constants/new
+  def new
+    @constant = Constant.new
+  end
+
+  # GET /constants/1/edit
+  def edit
+  end
+
+  # POST /constants
+  # POST /constants.json
+  def create
+    @constant = Constant.new(constant_params)
+
+    respond_to do |format|
+      if @constant.save
+        format.html { redirect_to @constant, notice: 'Constant was successfully created.' }
```

Comment:
Line is too long. [91/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.